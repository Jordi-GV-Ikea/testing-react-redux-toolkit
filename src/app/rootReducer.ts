import { combineReducers } from '@reduxjs/toolkit';
import profileReducer from '../features/profile/profileSlice';
import toDoReducer from '../features/toDo/toDoSlice';

const rootReducer = combineReducers({
  profile: profileReducer,
  toDo: toDoReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
