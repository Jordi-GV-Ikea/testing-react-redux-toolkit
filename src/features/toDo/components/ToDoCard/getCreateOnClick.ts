import { store } from 'app/store';
import { createToDo, showErrorMessage } from '../../toDoSlice';

export const getCreateOnClick = (profileId?: number) => (): void => {
  if (!profileId) {
    store.dispatch(showErrorMessage('Must select a profile first.'));
    return;
  }
  store.dispatch(createToDo(profileId));
};
