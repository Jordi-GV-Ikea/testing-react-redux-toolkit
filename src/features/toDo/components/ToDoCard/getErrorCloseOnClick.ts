import { store } from 'app/store';
import { hideErrorMessage } from '../../toDoSlice';

export const getErrorCloseOnClick = () => (): void => {
  store.dispatch(hideErrorMessage());
};
