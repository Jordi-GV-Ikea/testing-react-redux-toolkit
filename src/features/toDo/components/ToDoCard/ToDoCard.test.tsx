import React from 'react';
import { render, screen, within, RenderResult } from '@testing-library/react';
import { Provider } from 'react-redux';

import { store } from 'app/store';
import { ToDoCard } from './ToDoCard';

const renderCard = (): RenderResult =>
  render(
    <Provider store={store}>
      <ToDoCard />
    </Provider>,
  );

test('renders the ToDoCard', () => {
  const { asFragment } = renderCard();
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByLabelText(/toDo card/i)).toBeInTheDocument();
  const toDoList = screen.getByRole('list');
  expect(toDoList).toBeInTheDocument();

  // we have to use queryAll* instead of getAllBy* in order to test non-existence.
  // see https://stackoverflow.com/questions/52783144/how-do-you-test-for-the-non-existence-of-an-element-using-jest-and-react-testing
  const toDoListItems = within(toDoList).queryAllByRole('listitem');
  expect(toDoListItems).toHaveLength(0);
});
