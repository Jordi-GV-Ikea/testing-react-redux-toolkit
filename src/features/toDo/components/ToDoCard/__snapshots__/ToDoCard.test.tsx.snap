// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ToDoCard 1`] = `
<DocumentFragment>
  <div
    aria-label="ToDo Card"
    class="card"
    role="complementary"
  >
    <div
      class="card-header"
    >
      <h5
        class="card-title"
      >
        ToDos 
      </h5>
    </div>
    <div
      class="card-body"
    >
      <ul
        class="list-group"
      />
    </div>
    <div
      class="card-footer"
    >
      <div
        class="wrapper"
      >
        <button
          class="btn btn-primary disabled"
        >
          Create New ToDo
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
